pytorch_fairseq_translation.md

```diff
@@ -0,0 +1,116 @@
+---
+layout: hub_detail
+background-class: hub-background
+body-class: hub
+title: Neural Machine Translation with Transformer
+summary: Transformer model, pre-trained for English-French and English-German translation.
+category: researchers
+image: fairseq_logo.png
+author: Facebook AI Research (fairseq)
+tags: [nlp]
+github-link: https://github.com/pytorch/fairseq/
+accelerator: cuda-optional
+order: 10
+---
+
+
+### Model Description
+
+The Transformer, introduced in the paper [Attention Is All You Need][1], is a
+powerful sequence-to-sequence modeling architecture capable of producing
+state-of-the-art machine translation systems.
+
+Recently, the fairseq team has explored large scale semi-supervised training
+with big mini-batches and large quantities of back-translated data, further
+improving translation quality over the original model. More details can be found
+in [this blog post][2].
+
+
+### English-to-French Translation
+
+To translate from English to French using the model from the paper [Scaling
+Neural Machine Translation][3]:
+
+```python
+>>> import torch
+>>> en2fr = torch.hub.load(
+...   'pytorch/fairseq',
+...   'transformer',
+...   model_name_or_path='transformer.wmt14.en-fr',
+...   data_name_or_path='.',
+...   tokenizer='moses',
+...   aggressive_dash_splits=True,
+...   bpe='subword_nmt',
+... )
+>>> print(en2fr.generate('Hello world!'))
+Bonjour à tous !
+```
+
+
+### English-to-German Translation
+
+Back-translation is an effective way of improving translation quality through
+semi-supervised learning. In the paper [Understanding Back-Translation at
+Scale][4], we back-translate over 200 million German sentences to use as
+additional training data. An ensemble of these models was the winning submission
+to the [WMT'18 English-German news translation competition][5].
+
+To translate from English to German using one of the models in the ensemble
+(note: the archive includes the full ensemble and is 16GB after uncompression,
+so will take some time to download and extract):
+
+```python
+>>> import torch
```

Comment:
Please remove the ">>>" in the code block, our CI greps all code blocks in this file and run it against latest pytorch release ;) Thanks!

Reply:
Fixing now